lib/cisco_node_utils/bgp.rb

```diff
@@ -351,6 +351,7 @@ def confederation_peers
     def confederation_peers=(should)
       # confederation peers are additive. So create a delta hash of entries
       # and only apply the changes
+      should = should.flatten
```

Comment:
Doesn't seem to hurt to flatten a non-nested array:

irb(main):001:0> [["1", "2", "3"]]
=> [["1", "2", "3"]]
irb(main):002:0> [["1", "2", "3"]].flatten
=> ["1", "2", "3"]
irb(main):003:0> ["1", "2", "3"].flatten
=> ["1", "2", "3"]

I've thrown a array (pun intended) of different inputs/strings/arrays at this - seems to be safe.

Will do something like the depth check if you insist?